### PR TITLE
Fix breakpoints being skipped during the ballerina debug process

### LINKED ISSUE #19981: Debug points are being skipped during the debug process
**Description:**
Debug points are skipped during the debug process in both of the vscode and intellij plugins, when debugging a single ballerina source file or a ballerina module. 

**Steps to reproduce:**

**Affected Versions:**
ballerina distribution version - 1.0.4

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
Fixes #19981.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java [hunk 1]
```diff
@@ -115,25 +116,26 @@ public void execute() {
...
-    
+
             Runtime.getRuntime().exit(1);
             return;
         }
-    
-        // enable remote debugging
+
+        // enable remote debugging.
         if (null != this.debugPort) {
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java]: Shall we pass debug port directly to RunExecutableTask then we do not need to set any system property. Also in RunExecutableTask we can check the null != this.debugPort add java agent args.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java [hunk 2]
```diff
@@ -151,13 +156,16 @@ private void runGeneratedExecutable(BLangPackage executableModule, BuildContext
...
         String initClassName = BFileUtil.getQualifiedClassName(executableModule.packageID.orgName.value,
-                                                               executableModule.packageID.name.value,
-                                                               MODULE_INIT_CLASS_NAME);
+                executableModule.packageID.name.value, MODULE_INIT_CLASS_NAME);
         try {
             List<String> commands = new ArrayList<>();
             commands.add("java");
             commands.add("-cp");
             commands.add(getClassPath(executableJar));
+            if (isInDebugMode) {
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: RunExecutableTask can have debugport pass through runcommand and then we can check that here.

**Comment by warunalakshitha** [comment]
Can we write a test for debugging stuff so it will not break next time?

**Comment by pahans** [comment]
@NipunaRanasinghe @kaviththiranga we might have to repeat changes done to `RunCommand`, `TestCommand` as well.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java]: Packerina implementation has used the same implementation(system property) for all the other commands (push,pull,search, etc) as well. WDYT?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java]: https://github.com/ballerina-platform/ballerina-lang/pull/20087#discussion_r351645513

**Comment by NipunaRanasinghe** [comment]
@pahans yeah you are right.. will incorporate the required changes.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java (hunk 1 above)
Comment (warunalakshitha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/RunCommand.java]: +1 use system property.

**Comment by NipunaRanasinghe** [comment]
@pahans test command implementation has not been changed and it still used the class-loading approach instead of running an executable jar. I tried for test command and it works fine :)

**Pull request merged**